test/e2e/logstash/pipeline_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build logstash || e2e
+
+package logstash
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"testing"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/logstash"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestPipelineConfigRefLogstash(t *testing.T) {
+	secretName := "ls-generator-pipeline"
+
+	pipelineSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: test.Ctx().ManagedNamespace(0),
+		},
+		StringData: map[string]string{
+			"pipelines.yml": `
+- pipeline.id: generator
+  pipeline.workers: 1
+  queue.drain: false
+  config.string: input { generator {} } filter { sleep { time => 10 } } output { stdout { codec => dots } }
+- pipeline.id: main
+  config.string: input { stdin{} } output { stdout{} }`,
+		},
+	}
+
+	before := test.StepsFunc(func(k *test.K8sClient) test.StepList {
+		return test.StepList{}.WithStep(test.Step{
+			Name: "Create pipeline secret",
+			Test: test.Eventually(func() error {
+				return k.CreateOrUpdateSecrets(pipelineSecret)
+			}),
+		})
+	})
+
+	name := "test-pipeline-ref"
+	b := logstash.NewBuilder(name).
+		WithNodeCount(1).
+		WithPipelinesConfigRef(commonv1.ConfigSource{
+			SecretRef: commonv1.SecretRef{
+				SecretName: secretName,
+			},
+		})
+
+	steps := test.StepsFunc(func(k *test.K8sClient) test.StepList {
+		return test.StepList{
+			b.CheckMetricsRequest(k,
+				logstash.Request{
+					Name: "pipeline [generator]",
+					Path: "/_node/pipelines/generator",
+				},
+				logstash.Want{
+					Status: "green",
+				}),
```

Comment:
nit, out of curiosity: why are we not checking `pipelines.generator.workers` as it is the case for the default pipeline here: https://github.com/elastic/cloud-on-k8s/pull/6480/files#diff-4c2ec435669b8407b1fef39e9ce03a145ae797732051e4bbcb42caf136d975c6R117-R120

```suggestion
				logstash.Want{
					Status: "green",
					Match:  map[string]string{"pipelines.generator.workers": "1"},
				}),
```

Reply:
Good catch! should check pipeline.* to ensure the pipeline works. The default number of workers is the number of CPU cores. I am going to replace `pipelines.*.workers: 2` with `pipelines.*.batch_size: 125`, which is a static number from `logstash.yml`